core/connection/manager.go

```diff
@@ -209,7 +210,19 @@ func (m *connectionManager) Connect(consumerID, accountantID identity.Identity,
 
 	providerID := identity.FromAddress(proposal.ProviderID)
 
-	channel := m.createP2PChannel(m.ctx, consumerID, providerID, proposal)
+	var channel p2p.Channel
+	if contact, err := p2p.ParseContact(proposal.ProviderContacts); err == nil {
+		channel, err = m.createP2PChannel(m.ctx, consumerID, providerID, proposal.ServiceType, contact)
+		if err != nil {
+			return fmt.Errorf("could not create p2p channel: %w", err)
```

Comment:
Note that now if proposals supports p2p it will try to use p2p or fail and will not fallback to dialog. Fallback to dialog will happen only if proposal doesn't support p2p.

Reply:
Thats fine imho. Most likely it will to setup transport then anyways.